packer/builder/azure/driver_restapi/targets/step_stop_vm.go

```diff
@@ -18,17 +23,16 @@ type StepStopVm struct {
 }
 
 func (s *StepStopVm) Run(state multistep.StateBag) multistep.StepAction {
-	reqManager := state.Get(constants.RequestManager).(*request.Manager)
+	client := state.Get(constants.RequestManager).(management.Client)
 	ui := state.Get(constants.Ui).(packer.Ui)
 
 	errorMsg := "Error Stopping Temporary Azure VM: %s"
 
 	ui.Say("Stopping Temporary Azure VM...")
 
-	requestData := reqManager.ShutdownRoles(s.TmpServiceName, s.TmpVmName)
-	err := reqManager.ExecuteSync(requestData)
-
-	if err != nil {
+	if err := retry.ExecuteAsyncOperation(client, func() (management.OperationId, error) {
```

Comment:
I might have an idea to get rid of those lambdas you pass to ExecuteAsyncOperation...

Have an instance of `client` in `retry` object (I'm sure you can find a way) and then you can do:

    retry. ExecuteAsyncOperation(vmClient.ShutdownRole(s.TmpServiceName, s.TmpVmName, s.TmpVmName))

because go can pass `(OperationId, error)` pair to a method takes `(OperationId, error)`.